library/std/src/sys/pal/unix/mod.rs

```diff
@@ -61,6 +61,28 @@ pub unsafe fn init(argc: isize, argv: *const *const u8, sigpipe: u8) {
     }
 
     unsafe fn sanitize_standard_fds() {
+        let mut opened_devnull = -1;
+        let mut open_devnull = || {
+            #[cfg(not(all(target_os = "linux", target_env = "gnu")))]
+            use libc::open as open64;
+            #[cfg(all(target_os = "linux", target_env = "gnu"))]
+            use libc::open64;
+
+            if opened_devnull != -1 {
+                if libc::dup(opened_devnull) != -1 {
```

Comment:
I guess we're assuming that any operations on these files aren't mutating the underlying state in a way that we care about? E.g., seeking won't work because /dev/null is empty, etc.?

I'm not convinced the original strategy was much of a problem (nor is the duplication IMO that bad - the code was certainly easier to follow, I think, previously) - I'm inclined to not merge this personally.

Reply:
There isn't state *to* mutate, beyond the seek offset, which is barely even *preserved* (writes don't affect it under Linux and FreeBSD, seeks don't affect it under Linux). Seeking anywhere doesn't do anything anyway. You'd have to really *try* to detect a difference and may not even be able to, so TTBOMK this is just a straight optimisation.